tinkerpop/src/main/java/uk/gov/gchq/gaffer/gafferpop/GafferPopGraph.java

```diff
@@ -0,0 +1,759 @@
+import uk.gov.gchq.koryphe.iterable.MappedIterable;
+
+import java.nio.charset.StandardCharsets;
+import java.nio.file.Path;
+import java.nio.file.Paths;
+import java.util.Arrays;
+import java.util.Collections;
+import java.util.HashMap;
+import java.util.Iterator;
+import java.util.LinkedList;
+import java.util.List;
+import java.util.Map;
+import java.util.concurrent.ConcurrentHashMap;
+
+/**
+ * A <code>GafferPopGraph</code> is an implementation of
+ * {@link org.apache.tinkerpop.gremlin.structure.Graph}.
+ * It wraps a Gaffer {@link Graph} and delegates all operations to it.
+ * In addition to the tinkerpop methods required there are methods to add edges
+ * query for adjacent vertices and to provide a {@link View} to filter out results.
+ */
+public class GafferPopGraph implements org.apache.tinkerpop.gremlin.structure.Graph {
+    public static final String GRAPH_ID = "gaffer.graphId";
+
+    /**
+     * Configuration key for a path to Gaffer store properties.
+     *
+     * @see uk.gov.gchq.gaffer.store.StoreProperties
+     */
+    public static final String STORE_PROPERTIES = "gaffer.storeproperties";
+
+    /**
+     * Configuration key for a string array of path to Gaffer schemas.
+     *
+     * @see Schema
+     */
+    public static final String SCHEMAS = "gaffer.schemas";
+
+    /**
+     * Configuration key for a string array of operation options.
+     * Each option should in the form: key:value
+     */
+    public static final String OP_OPTIONS = "gaffer.operation.options";
+
+    public static final String USER_ID = "gaffer.userId";
+
+    public static final String DATA_AUTHS = "gaffer.dataAuths";
+
+    /**
+     * The vertex label for vertex IDs. These are {@link GafferPopVertex}s that
+     * don't have any properties, just an ID value and a label of 'id'.
+     */
+    public static final String ID_LABEL = "id";
+
+    private final Graph graph;
+    private final Configuration configuration;
+    private final GafferPopGraphVariables variables;
+    private final GafferPopGraphFeatures features;
+    private final Map<String, String> opOptions;
+    private final User user;
+    private final ServiceRegistry serviceRegistry;
+
+    private static final Logger LOGGER = LoggerFactory.getLogger(GafferPopGraph.class);
+
+    public GafferPopGraph(final Configuration configuration) {
+        this(configuration, createGraph(configuration));
+    }
+
+    public GafferPopGraph(final Configuration configuration, final Graph graph) {
+        this.configuration = configuration;
+        this.graph = graph;
+        features = new GafferPopGraphFeatures();
+        opOptions = new HashMap<>();
+        if (configuration().containsKey(OP_OPTIONS)) {
+            for (final String option : configuration().getStringArray(OP_OPTIONS)) {
+                final String[] parts = option.split(":");
+                opOptions.put(parts[0], parts[1]);
+            }
+        }
+
+        user = new User.Builder()
```

Comment:
Presumably other user types are not supported?

Reply:
I think right now yes, users are not implemented. I think it would be easiest to point to a ProxyStore and let the other graph figure out auths.

Reply:
Ok, might want to put that in the README as a limitation?